File: src/main/java/com/gmail/borlandlp/minigamesdtools/conditions/examples/EmptyInventoryCondition.java
package com.gmail.borlandlp.minigamesdtools.conditions.examples;

import com.gmail.borlandlp.minigamesdtools.conditions.AbstractCondition;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

public class EmptyInventoryCondition extends AbstractCondition {
    @Override
    public boolean isValidPlayer(Player p) {
        for (ItemStack itemStack : p.getInventory().getContents()) {
            if(itemStack != null && itemStack.getType() != Material.AIR) {
                return false;
            }
        }

        return true;
    }

    @Override
    public String getErrorId() {
        return "non_empty_inventory_error";
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/conditions/examples/EmptyInventoryCondition.kt
package com.gmail.borlandlp.minigamesdtools.conditions.examples

import com.gmail.borlandlp.minigamesdtools.conditions.AbstractCondition
import org.bukkit.Material
import org.bukkit.entity.Player

class EmptyInventoryCondition : AbstractCondition() {
    override fun isValidPlayer(player: Player): Boolean {
        for (itemStack in player.inventory.contents) {
            if (itemStack?.type != Material.AIR) {
                return false
            }
        }
        return true
    }

    override val errorId: String
        get() = "non_empty_inventory_error"
}